src/main/java/tictac7x/charges/items/U_ColossalPouch.java

```diff
@@ -0,0 +1,132 @@
+            // Empty
+            new OnChatMessage("There is no essence in this pouch.").emptyStorage(),
+            new OnChatMessage("The rift becomes active!").emptyStorage(),
+
+            // Check
+            new OnChatMessage("There (?:is|are) (?<charges>.+?) (?<essence>pure|daeyalt|guardian)? ?essences? in this pouch.").matcherConsumer((m) -> {
+                storage.clear();
+
+                int itemID;
+                switch (m.group("essence")) {
+                    case "pure":
+                        itemID = ItemID.PURE_ESSENCE;
+                        break;
+                    case "daeyalt":
+                        itemID = ItemID.DAEYALT_ESSENCE;
+                        break;
+                    case "guardian":
+                        itemID = ItemID.GUARDIAN_ESSENCE;
+                        break;
+                    default:
+                        itemID = ItemID.RUNE_ESSENCE;
+                        break;
+                }
+                
+                storage.put(itemID, getNumberFromWordRepresentation((m.group("charges"))));
+            }).onMenuOption("Check"),
+
+            // Decay.
+            new OnChatMessage("Your pouch has decayed through use.").onMenuOption("Fill").consumer(() -> {
+                this.decayCount += 1;
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+            
+            // Repair.
+            new OnChatMessage( "Fine. A simple transfiguration spell should resolve things for you.").consumer(() -> {
+                this.decayCount = 0;
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+
+            // Fill from inventory.
+            new OnItemContainerChanged(ItemContainerId.INVENTORY).fillStorageFromInventorySingle().onMenuOption("Fill"),
+
+            // Fill from bank.
+            new OnItemContainerChanged(ItemContainerId.BANK).fillStorageFromInventorySingle().onMenuOption("Fill"),
+
+            // Use essence on pouch.
+            new OnItemContainerChanged(INVENTORY).fillStorageFromInventorySingle().onUseStorageItemOnChargedItem(storage.getStoreableItems()),
+
+            // Empty to inventory.
+            new OnItemContainerChanged(ItemContainerId.INVENTORY).emptyStorageToInventory().onMenuOption("Empty"),
+
+            // Set maximum charges on level up
+            new OnStatChanged(Skill.RUNECRAFT).consumer(() -> {
+                this.storage.maximumTotalQuantity(getMaxCharges());
+            }),
+        };
+    }
+
+    public int getMaxCharges() {
+        int runecraftLevel = this.client.getRealSkillLevel(Skill.RUNECRAFT);
+        if (runecraftLevel >= 85) {
+            return 40 - (decayCount * 5);
+        }
+        else if (runecraftLevel >= 75) {
+            return 27 - (decayCount * 3);
+        }
+        else if (runecraftLevel >= 50) {
+            return 16 - (decayCount * 2);
+        }
+        else return 8;
```

Comment:
Unsure if the lowest level degrades. If it does, I assume it would be by `1`.

Reply:
Wiki states that it does yes and the pouch is destroyed in the process